test/test_circ_deque.jl

```diff
@@ -78,6 +78,11 @@ end
     empty!(D)
     unshift!(D, 40)
     @test front(D) == back(D) == 40
+
+    # Test iteration over loop
+    D = CircularDeque{Int}(5)
+    for i in 1:5 push!(D, i) end
+    @test collect(i for i in D) == collect(1:5)
```

Comment:
Generators aren't available on Julia 0.4.  To maintain compatibility, you could enclose the generator inside an array:

```julia
collect([i for i in D])
```